src/_lib/modules/word-cloud/word-cloud.component.ts

```diff
@@ -0,0 +1,433 @@
+    private _ctx: CanvasRenderingContext2D;
+    private _centerPoint: IPoint;
+    /**
+     * Controls how many points are generated on the spiral. Should be kept between 0 and 1. Lower values generate more
+     * points on the spiral but it can take a lot longer to position the words. If updating this value to <= 0.3, consider
+     * all the places where this is used and for each one the amount of words in the cloud. It could also be a good idea to
+     * make this value configurable by allowing it to be specified in the `options` input
+     */
+    private _spiralResolution = 0.3;
+
+    public config: IWordCloudConfig;
+
+    constructor(
+        private _elRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        private _cdRef: ChangeDetectorRef,
+        @Inject(DOCUMENT) private _document: Document,
+        public elRef: ElementRef<HTMLElement>) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const wordsChange: boolean = changes.words?.currentValue !== changes.words?.previousValue;
+        const optionsChange: boolean = changes.options?.currentValue !== changes.options?.previousValue;
+
+        if (wordsChange || optionsChange) {
+            this._init();
+        }
+    }
+
+    /**
+     * Exports the canvas to a PNG image and returns the base64-encoded PNG data
+     * @returns A string containing the base64-encoded PNG data of the canvas
+     */
+    exportCanvas(): string {
+        /**
+         * Create a new canvas matching the dimensions of the original canvas. At this point, `_positionedWords` contains
+         * the final positions of the words and we have no need to check for intersections, so we loop through our words
+         * and draw them on our `exportCanvas`
+         */
+        const exportCanvas = this._renderer.createElement('canvas') as HTMLCanvasElement;
+        const exportCtx = exportCanvas.getContext('2d');
+        exportCanvas.width = this._canvas.width;
+        exportCanvas.height = this._canvas.height;
+
+        this._positionedWords.forEach(pw => {
+            const point: IPoint = { x: pw.canvasX, y: pw.canvasY };
+            this._setFontDetails(exportCtx, pw.fontSize);
+            this._drawWord(pw, point, exportCtx);
+        });
+
+        return exportCanvas.toDataURL('image/png');
+    }
+
+    downloadCanvas(filename: string): void {
+        const dataUrl = this.exportCanvas();
+        const link = document.createElement('a');
+
+        link.download = `${filename}.png`;
+        link.href = dataUrl;
+        link.click();
+    }
+
+    onResize() {
+        this._init();
+    }
+
+    private _init(): void {
+        this.config = this._getConfig();
+        this._truncatedWordsWithFontSize = this._getTruncatedWordsWithFontSize(this.words);
+        this._positionedWords = [];
+        this._drawCanvas();
+        this._centerPoint = { x: this._canvas.width / 2, y: this._canvas.height / 2 };
+
+        this._isFontLoaded().subscribe(() => {
```

Comment:
Does this need to be destroyed?

Reply:
For the sake of not adding adding a `destroyed$` subject and the associated logic in `ngOnDestroy`, I've left it out as the osbervable is piped with `take(1)` so as soon as the observable emits it will complete